src/AspNet.Security.OAuth.LinkedIn/LinkedInAuthenticationHandler.cs

```diff
@@ -33,12 +34,13 @@ public class LinkedInAuthenticationHandler : OAuthHandler<LinkedInAuthentication
             [NotNull] AuthenticationProperties properties, [NotNull] OAuthTokenResponse tokens)
         {
             var address = Options.UserInformationEndpoint;
+            var fields = Options.Fields.Where(f => f != LinkedInAuthenticationConstants.EmailAddressField).ToList();
 
             // If at least one field is specified,
             // append the fields to the endpoint URL.
-            if (Options.Fields.Count != 0)
+            if (fields.Count != 0)
             {
-                address = address.Insert(address.LastIndexOf("~") + 1, $":({ string.Join(",", Options.Fields)})");
+                address = $"{address}?projection=({string.Join(",", fields)})";
```

Comment:
If for some reason `UserInformationEndpoint` already has a query string in it, this would break because it would try and add another. You should use a `UriBuilder` or the `AddQueryString(...)` helper used elsewhere to append the extra query string parameter.

Reply:
Good point, I'm using `AddQueryString(...)` now